ext/dcos-installer/dcos_installer/action_lib.py

```diff
@@ -196,31 +242,20 @@ def install_dcos(config, block=False, state_json_dir=None, hosts=None, async_del
         hosts = []
     assert isinstance(hosts, list)
 
-    # Role specific parameters
-    role_params = {
-        'master': {
-            'tags': {'role': 'master', 'dcos_install_param': 'master'},
-            'hosts': config['master_list']
-        },
-        'agent': {
-            'tags': {'role': 'agent', 'dcos_install_param': 'slave'},
-            'hosts': config.get('agent_list', [])
-        },
-        'public_agent': {
-            'tags': {'role': 'public_agent', 'dcos_install_param': 'slave_public'},
-            'hosts': config.get('public_agent_list', [])
-        }
-    }
-
     bootstrap_tarball = _get_bootstrap_tarball()
     log.debug("Local bootstrap found: %s", bootstrap_tarball)
 
     targets = []
     if hosts:
         targets = hosts
     else:
-        for role, params in role_params.items():
-            targets += [Node(node, params['tags']) for node in params['hosts']]
+        # state_json_dir contains a path to installer states. If it is not empty, it means we have executed
+        # preflight state already and we want to take only the hosts which passed the preflight. Otherwise get a list
+        # of hosts from a config file (used in CLI where the stages do not have to be in order).
+        if state_json_dir is None:
+            targets = get_full_nodes_list(config)
```

Comment:
If I have a config.yaml, do an install, then make / edit config.yaml and do another install, what will happen here? Will it pick up the state_json_dir from my first install and try resuming that or will it pick up my new config?